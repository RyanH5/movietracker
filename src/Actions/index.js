export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
});

export const toggleUserLogin = (userData) =>({
  type: 'TOGGLE_USER_LOGIN',
  userData
});