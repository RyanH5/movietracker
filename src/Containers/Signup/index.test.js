import {Signup, mapStateToProps, mapDispatchToProps} from './index';
import { shallow, mount } from 'enzyme';
import React from 'react';
import { toggleUserLogin, userIsFalse, userSignup } from '../../Actions';

describe('Login', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(<Signup />);

    expect(wrapper).toMatchSnapshot();
  });
  
  describe('handleChange', () => {
    let spy;
    let wrapper;

    beforeEach(() => {
      wrapper = mount(<Signup />);
      spy = jest.spyOn(wrapper.instance(), 'handleChange');
    });

    it('should update state of email', () => {
      const mockEvent = {
        target: {
          value: 'h',
          name: 'email'
        }
      };
      wrapper.instance().handleChange(mockEvent);
      wrapper.find('input').first().simulate('change'); 

      expect(spy).toHaveBeenCalled();
    });

    it('should update state of password', () => {
      const mockEvent = {
        target: {
          value: 'h',
          name: 'password'
        }
      };
      wrapper.instance().handleChange(mockEvent);
      wrapper.find('input').last().simulate('change'); 

      expect(spy).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('should create a props object with the correct keys', () => {
      const mockState = {
        user: {
          loginStatus: true
        }
      };
      const expected = {
        loginStatus: true
      };
      const mappedProps = mapStateToProps(mockState);
  
      expect(mappedProps).toEqual(expected);
  
    });
  }); 

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using toggleUserLogin from MDTP', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = toggleUserLogin({id: 3, loginStatus: true});
      const mappedToProps = mapDispatchToProps(mockDispatch);
      mappedToProps.toggleUserLogin({id: 3, loginStatus: true});
      
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  
    it('should call dispatch when using userIsFalse from MDTP', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = userIsFalse({id: 3, loginStatus: true});
      const mappedToProps = mapDispatchToProps(mockDispatch);
      mappedToProps.userIsFalse({id: 3, loginStatus: true});
      
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  
    it('should call dispatch when using userSignup from MDTP', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = userSignup([{id: 3, name: 'Jill', loginStatus: false}]);
      const mappedToProps = mapDispatchToProps(mockDispatch);
      mappedToProps.userSignup([{id: 3, name: 'Jill', loginStatus: false}]);
      
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

  
});

