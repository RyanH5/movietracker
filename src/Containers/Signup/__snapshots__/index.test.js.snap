// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <h2>
          SIGNUP
</h2>,
        <input
          name="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value=""
/>,
        <input
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="text"
          value=""
/>,
        <input
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="text"
          value=""
/>,
        <button
          className="sign-up"
>
          Sign Up
</button>,
      ],
      "className": "sign-up-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SIGNUP",
        },
        "ref": null,
        "rendered": "SIGNUP",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "name",
          "onChange": [Function],
          "placeholder": "Name",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "email",
          "onChange": [Function],
          "placeholder": "email",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "password",
          "onChange": [Function],
          "placeholder": "password",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "className": "sign-up",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <h2>
            SIGNUP
</h2>,
          <input
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
/>,
          <input
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="text"
            value=""
/>,
          <input
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="text"
            value=""
/>,
          <button
            className="sign-up"
>
            Sign Up
</button>,
        ],
        "className": "sign-up-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SIGNUP",
          },
          "ref": null,
          "rendered": "SIGNUP",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "onChange": [Function],
            "placeholder": "Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "className": "sign-up",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
